src/gas/gas-util.test.ts

```diff
@@ -155,72 +227,741 @@ describe('gas utils', () => {
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...low,
+        },
+        medium: {
+          minWaitTimeEstimate: 0,
+          maxWaitTimeEstimate: 0,
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...medium,
+        },
+        high: {
+          minWaitTimeEstimate: 0,
+          maxWaitTimeEstimate: 0,
+          suggestedMaxPriorityFeePerGas: '0',
+          suggestedMaxFeePerGas: '0',
+          ...high,
+        },
+        estimatedBaseFee,
+        historicalBaseFeeRange: null,
+        baseFeeTrend: null,
+        latestPriorityFeeRange: null,
+        historicalPriorityFeeRange: null,
+        priorityFeeTrend: null,
+        networkCongestion: null,
+      };
+    };
 
-    it('should handle inputs with more than 9 decimal places', () => {
-      expect(normalizeGWEIDecimalNumbers(1.0000000162)).toBe('1.000000016');
-      expect(normalizeGWEIDecimalNumbers(1.0000000165)).toBe('1.000000017');
-      expect(normalizeGWEIDecimalNumbers(1.0000000199)).toBe('1.00000002');
-      expect(normalizeGWEIDecimalNumbers(1.9999999999)).toBe('2');
-      expect(normalizeGWEIDecimalNumbers(1.0000005998)).toBe('1.0000006');
-      expect(normalizeGWEIDecimalNumbers(123456.0000005998)).toBe(
-        '123456.0000006',
-      );
-      expect(normalizeGWEIDecimalNumbers(1.000000016025)).toBe('1.000000016');
-      expect(normalizeGWEIDecimalNumbers(1.0000000160000028)).toBe(
-        '1.000000016',
-      );
-      expect(normalizeGWEIDecimalNumbers(1.000000016522)).toBe('1.000000017');
-      expect(normalizeGWEIDecimalNumbers(1.000000016800022)).toBe(
-        '1.000000017',
-      );
-    });
+    describe('if the given priority fee is less than the given max fee minus the latest base fee', () => {
+      let maxPriorityFeePerGas: string;
+      let maxFeePerGas: string;
+      let estimatedBaseFee: string;
+
+      beforeEach(() => {
+        maxPriorityFeePerGas = '1';
+        maxFeePerGas = '102';
+        estimatedBaseFee = '100';
+      });
+
+      describe('and the given priority fee does not reach any of the suggested priority fee thresholds', () => {
+        it('should return no lower bound and an unknown upper bound', () => {
+          const timeBounds = calculateTimeEstimate(
+            maxPriorityFeePerGas,
+            maxFeePerGas,
+            buildGasFeeEstimates({
+              estimatedBaseFee,
+              low: {
+                suggestedMaxPriorityFeePerGas: (
+                  Number(maxPriorityFeePerGas) + 0.5
+                ).toString(),
+              },
+            }),
+          );
+
+          expect(timeBounds).toStrictEqual({
+            lowerTimeBound: null,
+            upperTimeBound: 'unknown',
+          });
+        });
+      });
+
+      describe('and the given priority fee reaches the suggested low priority fee threshold, but does reach the medium priority fee threshold', () => {
```

Comment:
Nit: This seemed more concise:

```suggestion
      describe('and the given priority fee is equal to the suggested low priority fee', () => {
```

and the same for the others, except replace "equal to" with "exceeds", or "is between the suggested __ and __ priority fee".

Reply:
Though, this wording does remind me, is it possible for the suggested low and medium priority fee to be identical? Maybe we should test that case as well.

Reply:
It is possible. That's the reason I used "threshold" in the names of the tests. But I see how that would be confusing. I can revisit this.

Reply:
Oh, regarding low and medium being equal, what I meant was that this circumstance was not covered by any existing test. I didn't think the current descriptions implied it was impossible or anything, just that we could add a test case to cover this. Or multiple I guess (one for low === medium, one for medium === high, and one for all three being equal).

Reply:
Yup I can do that!